operator/internal/status/storage.go

```diff
@@ -25,5 +42,47 @@ func SetStorageSchemaStatus(ctx context.Context, k k8s.Client, req ctrl.Request,
 		Schemas: schemas,
 	}
 
+	for _, sc := range schemas {
+		date, err := sc.EffectiveDate.UTCTime()
+		if err != nil {
+			return kverrors.Wrap(err, "failed to parse effective date")
+		}
+		if sc.Version == lokiv1.ObjectStorageSchemaV13 && date.Before(cutoff) {
+			schemaVersionMap[sc.Version] = true
+			continue
+		} else {
+			schemaVersionMap[sc.Version] = false
+			oldSchemas = append(oldSchemas, sc)
+		}
+	}
+
+	if len(oldSchemas) > 0 {
+		if schemaVersionMap[lokiv1.ObjectStorageSchemaV13] {
+			//s.Status.Storage.Schemas[] = StorageSchemaOutOfRetention
+			if err := updateSchemaStatus(s.Status.Storage.Schemas, StorageSchemaOutOfRetention); err != nil {
```

Comment:
I think this can be done easier by going through the slice of schemas once and setting the matching status on it.

Reply:
The reason I did this is because the status `StorageSchemaOutOfRetention` is only activated if we have a newer v13 schema added whose effective date is in the past.